source/Conversation.cpp

```diff
@@ -298,6 +317,25 @@ Conversation Conversation::Substitute(const map<string, string> &subs) const
 	for(Node &node : result.nodes)
 		for(pair<string, int> &choice : node.data)
 			choice.first = Format::Replace(choice.first, subs);
+	
+	return result;
+}
+
+
+
+// Do text replacement throughout this conversation, and instantiate any
+// potential actions.
+Conversation Conversation::Instantiate(map<string, string> &subs, int jumps, int payload) const
```

Comment:
Do we need both the Instantiate and the Substitute function?
If Instantiate is the better/more-complete version that can completely replace Substitute, then please just remove Substitute.

Reply:
Instantiate is only called by MissionAction, but Substitute gets called at other points in the code. (I forget exactly where, but I did run into them when I tried to only have one function. Substitute also takes in a const map, while Instantiate is just a map, which was important for those places where Substitute is still used now.)

Reply:
I would expect that `subs` could also be given as const-map in Instantiate, because I would expect Instantiate to never change data in the map itself.

Other uses of Substitute that I just found are in main.cpp, MissionAction.cpp, NPC.cpp and I don't really see a reason for them to perform text-substitutes but not action-initialisation (with some sensible defaults).

Reply:
I’ll look into those other instances to see if Substitute can be entirely replaced.

Reply:
It looks like the ones in NPC should be Instantiate, but I'm not sure what's going on with the LoadConversation function in main. Looks like it's some sort of launch option that gets you there. That could be Instantiate as well, since that doesn't actually affect the game.